lib/tools/helpers/web-application-firewall/quotas-and-capacity.ts

```diff
@@ -593,7 +593,16 @@ function calculateRatebasedStatementwithoutScopeDownStatement(customRule: FmsRul
    * @returns the capacity of the IPSet statement
    */
 function calculateIpsSetStatementCapacity(ipSetReferenceStatement: wafv2.CfnWebACL.IPSetReferenceStatementProperty) {
-  let ipSetRuleCapacity = 1;
+  // FIXME: ipset statements doesn't always has a capacity of one, it can vary, which was giving a error while deploying (sample of the error available below)
```

Comment:
Hi, according to the docs an IPSet reference statement is only 4 if you configure the statement to use forwarded IP addresses and specify a position of ANY, increase the WCU usage by 4.

https://docs.aws.amazon.com/waf/latest/developerguide/waf-rule-statement-type-ipset-match.html

Could you please give me an example rule which was not working, maybe we need to ask AWS to change their docs?

Reply:
Hi @daknhh! It says in the docs which you've sent that the cost is 1 WCU **for most**, not for all cases.

![image](https://github.com/globaldatanet/aws-firewall-factory/assets/16310611/665e8d9e-b8aa-4aa4-80c9-4f4fa986b278)

I think there's some edge cases that the cost of a simple single IpsetReferenceStatement can be bigger then 1 WCU. In my FMS configs the error wasn't happening on all other configs, but on a single one which has a `Scope` of `CLOUDFRONT`.

I don't know if the scope and/or the `Type` of `AWS::CloudFront::Distribution` has anything to do with the bug, with the WCU cost being higher then "1", but I know that WAF started deploying after I've set a fixed WCU cost of 4 for single IpSetStatements, hence this commit.

I've suggested a long term solution for this issue in the comments on the commit.

Reply:
I will also open a support case on Monday to clarify with AWS support. thanks.

Reply:
Still waiting for Feedback from AWS Support. I am on it :)